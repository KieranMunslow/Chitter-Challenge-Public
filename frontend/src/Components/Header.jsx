import { Navbar, Nav, Container, Form, FormControl, Button, FormLabel, NavDropdown } from 'react-bootstrap';

const Header = ({ loggedIn, username }) => {

    let nav;

    if (loggedIn) {
        nav = (
            <Navbar collapseOnSelect expand="lg" bg="dark" variant="dark">
                <Container>
                    <Navbar.Brand href="#home">Chitter</Navbar.Brand>
                    <Navbar.Toggle aria-controls="responsive-navbar-nav" />
                    <Navbar.Collapse id="responsive-navbar-nav">
                        <Nav className="me-auto">
                            <Navbar.Text>
                                Signed in as: @{username}
                            </Navbar.Text>
                        </Nav>
                        <Nav>
                            <Nav.Link href="#deets">View Peeps</Nav.Link>
                            <Nav.Link eventKey={2} href="#memes">
                                Add Peep
                            </Nav.Link>
                            <Nav.Link>Logout</Nav.Link>
                        </Nav>
                    </Navbar.Collapse>
                </Container>
            </Navbar>
        );
    } else {
        nav = <Navbar collapseOnSelect expand="lg" bg="dark" variant="dark">
            <Container>
                <Navbar.Brand href="#home">Chitter</Navbar.Brand>
                <Navbar.Toggle aria-controls="responsive-navbar-nav" />
                <Navbar.Collapse id="responsive-navbar-nav">
                    <Nav className="me-auto"></Nav>
                    <Nav>
                        <Nav.Link href="#deets">View Peeps</Nav.Link>
                        <Nav.Link eventKey={2} href="#memes">
                            Register
                        </Nav.Link>
                        <Nav.Link>Login</Nav.Link>
                    </Nav>
                </Navbar.Collapse>
            </Container>
        </Navbar>
    }

    return (
        <>
            {nav}
        </>
    )
}

export default Header;