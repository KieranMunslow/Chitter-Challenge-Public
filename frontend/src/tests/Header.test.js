import { render } from '@testing-library/react';
import Header from '../Components/Header';


describe(`Tests for the header of the app`, () => {
    test(`Header matches snapshot when not logged in`, () => {
        expect(render(<Header loggedIn={false} username={``} />)).toMatchSnapshot();
    });

    test(`Header matches snapshot when logged in`, () => {
        expect(render(<Header loggedIn={true} username={`Name`} />)).toMatchSnapshot();
    });
});