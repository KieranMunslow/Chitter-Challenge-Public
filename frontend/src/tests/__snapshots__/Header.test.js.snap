// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the header of the app Header matches snapshot when logged in 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-lg navbar-dark bg-dark"
      >
        <div
          class="container"
        >
          <a
            class="navbar-brand"
            href="#home"
          >
            Chitter
          </a>
          <button
            aria-controls="responsive-navbar-nav"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="responsive-navbar-nav"
          >
            <div
              class="me-auto navbar-nav"
            >
              <span
                class="navbar-text"
              >
                Signed in as: @
                Name
              </span>
            </div>
            <div
              class="navbar-nav"
            >
              <a
                class="nav-link"
                data-rr-ui-event-key="#deets"
                href="#deets"
              >
                View Peeps
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="2"
                href="#memes"
              >
                Add Peep
              </a>
              <a
                class="nav-link"
                role="button"
                tabindex="0"
              >
                Logout
              </a>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="#home"
        >
          Chitter
        </a>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          >
            <span
              class="navbar-text"
            >
              Signed in as: @
              Name
            </span>
          </div>
          <div
            class="navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="#deets"
              href="#deets"
            >
              View Peeps
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="2"
              href="#memes"
            >
              Add Peep
            </a>
            <a
              class="nav-link"
              role="button"
              tabindex="0"
            >
              Logout
            </a>
          </div>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tests for the header of the app Header matches snapshot when not logged in 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-lg navbar-dark bg-dark"
      >
        <div
          class="container"
        >
          <a
            class="navbar-brand"
            href="#home"
          >
            Chitter
          </a>
          <button
            aria-controls="responsive-navbar-nav"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="responsive-navbar-nav"
          >
            <div
              class="me-auto navbar-nav"
            />
            <div
              class="navbar-nav"
            >
              <a
                class="nav-link"
                data-rr-ui-event-key="#deets"
                href="#deets"
              >
                View Peeps
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="2"
                href="#memes"
              >
                Register
              </a>
              <a
                class="nav-link"
                role="button"
                tabindex="0"
              >
                Login
              </a>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="#home"
        >
          Chitter
        </a>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          />
          <div
            class="navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="#deets"
              href="#deets"
            >
              View Peeps
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="2"
              href="#memes"
            >
              Register
            </a>
            <a
              class="nav-link"
              role="button"
              tabindex="0"
            >
              Login
            </a>
          </div>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
